docs/streaming.md

```diff
@@ -70,10 +70,13 @@ Disabling stderr and streaming only stdout into log file.
 
 # Guarantees
 
-When a task is *finished* or *failed* with a non-streaming error then it is guaranteed that its stream is fully flushed into the log file.
+When a task is *finished*, *failed* with a non-streaming error then it is guaranteed that its stream is fully flushed into the log file.
 
-When a task is *canceled* or *failed* with a streaming error, then the stream is not necessarily fully written into the log file in the moment when the state occurs
- and some part may be written later, but the stream will be eventually closed. In this case, HQ is also allowed to drop any suffix of the buffered part of the stream.
+When a task is *canceled* or *failed* with a streaming error, then the stream is not necessarily fully written into the log file in the moment when the state occurs and some part may be written later, but the stream will be eventually closed.
+
+When a task is *canceled*, *failed* because of its time limit, or streaming error
+then all task activities are terminated immediately including streaming and
+a suffix of the stream buffered in the worker may be lost.
```

Comment:
These sentences don't make sense. Both contain `canceled` and `streaming error` situations, but with different outocmes.

Reply:
They speak about two different things, the first is about the state of log file
The second one says that some suffix may be dropped